cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java

```diff
@@ -1159,12 +1159,17 @@ private void initNewClient(ModelControllerClient newClient, ControllerAddress ad
             }
 
             List<String> nodeTypes = Util.getNodeTypes(newClient, new DefaultOperationRequestAddress());
-            domainMode = nodeTypes.contains(Util.SERVER_GROUP);
+            // this is present even if the host hasn't been added yet.
+            domainMode = nodeTypes.contains(Util.HOST);
 
-            try {
-                extLoader.loadHandlers(currentAddress);
-            } catch (CommandLineException e) {
-                printLine(Util.getMessagesFromThrowable(e));
+            // if we're going to add a host manually, don't try to read the extensions,
+            // as they won't be there until after the /host:add()
+            if (! (nodeTypes.size() == 1 && nodeTypes.get(0).equals(Util.HOST))) {
```

Comment:
Just a note for the historical record:

AFAICT (could be wrong in which case this is just noise) this won't get called again following a reload, so starting an empty host config, then adding a host, and then reloading will not result in the CLI trying again to load extension handlers. I think this is ok; simple work around is to stop the embedded host and then start it again. If this proves to be an issue we can look into addressing it in the future.